### ISSUE TITLE
`convert` with shallow-copy data arrays

### ISSUE THREAD

**Issue by SimonHeybrock** [scipp/scippneutron#66]
The next release of `scipp` introduces a Pythonic shallow-copy mechanism for the major data structures. This discussion is to establish how we want to use and handle this in `convert`.

Most operations in scipp such as `+` (`__add__`) create new *data* (and masks) and share coords and attrs. Inplace operations such as `+=` (`__iadd__`) and `out` arguments are provided that can be used to avoid new allocations.

`convert` does not operate on data but on coordinates. I would propose the following mechanism:

1. Share data and untouched metadata by default, but create a new coord for the output:
   ```python
   # tof.data and wavelength.data reference same buffer
   # wavelength.coords['wavelength'] is a NEW buffer
   wavelength = scn.convert(tof, 'tof', 'wavelength', scatter=True)  
   ```
   As a bonus, this will allow for adding a keyword argument for preserving the original coord in the future.
2. If a user whishes to avoid sharing the can use `copy()`:
   ```python
   wavelength = scn.convert(tof, 'tof', 'wavelength', scatter=True).copy()
   # or
   wavelength = scn.convert(tof.copy(), 'tof', 'wavelength', scatter=True)
   ```
   Note that this makes two copies for the coord, but for this application this may be irrelevant at this point. We can also just copy data to avoid this:
   ```python
   wavelength = scn.convert(tof, 'tof', 'wavelength', scatter=True)
   wavelength.data = wavelength.data.copy()
   ```
3. Full in-place operation avoiding also the copy of coords should rely on `out` args as it is now. The current mechanism only supports `out==in` since ensuring that the dims and metadata layout match is difficult:
   ```python
   wavelength = scn.convert(tof, 'tof', 'wavelength', scatter=True, out=tof)
   ```
   On the C++ side we can now no longer rely on moving the input in this case, since this cannot be used for (buffer) ownership transfer given that there can now be multiple references to a buffer. We could add an `inplace` flag to the C++ implementation of `convert`, but this does not feel like a nice solution. Looking at the bigger picture we see that 1.) will actually provide a substantial speedup compared to the current implementation:
   - Currently (without `out` arg) data and other coords are copied.
   - This is most relevant for event data, where in a typicaly case we have 4 columns in the event data (typically 64 bit TOF, 64 bit pulse_time, 32+32 bit weights with uncertainties).
   - Based on the new sharing data structures we would replace only one of the 64 bit columns (`tof` -> `wavelength`), thus giving a substantial speedup.

In conclusion:
- Implement 1.)
- Do not implemented 3.) for now. We can maybe keep the `out` arg but not use it?
- Keep thinking about nicer solutions for 3.). Would it make sense to support this differently, e.g. using `event_out=tof.bins.coords['tof']`? By specifying a specific variable/buffer we can then also support this more generally (not just with `out=in`), and the intention is cleared.
